packages/gatsby/src/query-runner/query-compiler.js

```diff
@@ -50,7 +50,8 @@ class Runner {
   }
 
   async parseEverything() {
-    let files = await globp(`${this.baseDir}/**/*.js`)
+    let files = await globp(`${this.baseDir}/**/*.+(t|j)s?(x)`)
+    files = files.filter(d => !d.match(/(\.stories\.|\.test\.|\.d\.)/))
```

Comment:
I understand why you chose it :) my point is only that conventions only meet some peoples needs. for instance in my own projects this wouldn't catch any of my test files or stories. More importantly tho it will likely silently filter real source files which is a different version of the problem of this problem